### PR TITLE
[youtube:tab] Extract channel banner

### PR DESCRIPTION
**Pull request by coletdjnz**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [X] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Extracts channels banners if available. They are set to a lower preference than the channel avatar. Also try get the original/uncropped banner as described in https://github.com/yt-dlp/yt-dlp/issues/2237#issuecomment-1013694714.

Closes https://github.com/yt-dlp/yt-dlp/issues/2237

### CODE REVIEW

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -3913,11 +3913,28 @@ def _extract_from_tabs(self, item_id, ytcfg, data, tabs):
...
                 primary_sidebar_renderer, ('thumbnailRenderer', 'playlistVideoThumbnailRenderer', 'thumbnail')))
 
+        channel_banners = self._extract_thumbnails(
+            data, ('header', ..., ['banner', 'mobileBanner', 'tvBanner']))
+        for banner in channel_banners:
+            banner['preference'] = -10
+
+        if channel_banners:
+            # We can get the uncropped banner by replacing the crop params with '=s0'
+            # See: https://github.com/yt-dlp/yt-dlp/issues/2237#issuecomment-1013694714
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:3928]: that works for profile-pic too, according to the comment

Comment (coletdjnz) [reply yt_dlp/extractor/youtube.py:3928]: thanks, missed that. I might as well add that too

**Comment by blackdimund** [comment]
any way to get these banners to show up in the json file from the --write-info-json option?

**Comment by coletdjnz** [comment]
> any way to get these banners to show up in the json file from the --write-info-json option?

It's in the playlist info json, under thumbnails. All but the highest priority one are banners of different qualities.

**Comment by blackdimund** [comment]
What channel URL did you give yt-dlp? I am using the latest source code, and was only able to get the user's avatar as one single entry in the thumbnail list, sometimes a few more thumbnail entries that are various sizes of the user's avatar. The test channel URL I am giving is `https://www.youtube.com/channel/UCyJ_Gr6K0cyunUhwWJh-KeA`, the command looks like this `/usr/bin/python3 yt-dlp-master/yt_dlp/__main__.py --output output_path --skip-download --write-info-json --retries 10 https://www.youtube.com/channel/UCyJ_Gr6K0cyunUhwWJh-KeA`
thumbnails

**Comment by coletdjnz** [comment]
> What channel URL did you give yt-dlp? I am using the latest source code, and was only able to get the user's avatar as one single entry in the thumbnail list, sometimes a few more thumbnail entries that are various sizes of the user's avatar. The test channel URL I am giving is `https://www.youtube.com/channel/UCyJ_Gr6K0cyunUhwWJh-KeA`, the command looks like this `/usr/bin/python3 yt-dlp-master/yt_dlp/__main__.py --output output_path --skip-download --write-info-json --retries 10 https://www.youtube.com/channel/UCyJ_Gr6K0cyunUhwWJh-KeA` thumbnails

I was using `yt-dlp ytuser:pewdiepie --write-info-j --flat-playlist --playlist-items 1`

They show up for me. double check that the yt-dlp that you are running is up to date?

**Comment by blackdimund** [comment]
The problem seems to have been using subprocess, after using os.system instead, it works. Not sure why.

**Pull request merged**